internal/db/mailbox.go

```diff
@@ -241,8 +244,8 @@ func TranslateRemoteMailboxIDs(ctx context.Context, client *ent.Client, mboxIDs
 	}), nil
 }
 
-func CreateMailboxIfNotExists(ctx context.Context, tx *ent.Tx, internalID imap.InternalMailboxID, mbox imap.Mailbox, delimiter string) error {
-	exists, err := MailboxExistsWithID(ctx, tx.Client(), internalID)
+func CreateMailboxIfNotExists(ctx context.Context, tx *ent.Tx, mbox imap.Mailbox, delimiter string) error {
+	exists, err := MailboxExistsWithRemoteID(ctx, tx.Client(), mbox.ID)
```

Comment:
is this changed because this used to be a bug?

Reply:
Since the mailbox ID's are auto generated in the database now, we don't have access to the internal ID, so the check has been replaced with the remote ID instead.